// angular-plotly.js/esm5/src/app/plotly/plotly.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import * as PlotlyJS from 'plotly.js/dist/plotly.js';
import { PlotComponent } from '../shared/plot/plot.component';
import { PlotlyService } from '../shared/plotly.service';
import { SharedModule } from '../shared/shared.module';
var PlotlyModule = /** @class */ (function () {
    function PlotlyModule() {
        PlotlyService.setPlotly(PlotlyJS);
    }
    /**
     * @return {?}
     */
    PlotlyModule.forRoot = /**
     * @return {?}
     */
    function () {
        return {
            ngModule: PlotlyModule,
            providers: [PlotlyService]
        };
    };
    PlotlyModule.decorators = [
        { type: NgModule, args: [{
                    imports: [CommonModule, SharedModule],
                    declarations: [],
                    exports: [PlotComponent]
                },] }
    ];
    /** @nocollapse */
    PlotlyModule.ctorParameters = function () { return []; };
    return PlotlyModule;
}());
export { PlotlyModule };
